tests/next_tests/integration_tests/feature_tests/test_util_cases.py

```diff
@@ -64,6 +65,43 @@ def test_allocate_const(cartesian_case):
     assert b == 42.0
 
 
+def test_tree_mapped_np_all_close_simple_arrays_close():
+    a = np.array([1.0, 2.0, 3.0])
+    b = a + 1e-05
+
+    assert tree_mapped_np_all_close(a, b) is True
+    assert tree_mapped_np_all_close(a, b, rtol=1e-06) is False
+
+
+def test_tree_mapped_np_all_close_simple_arrays_different():
+    a = np.array([1.0, 2.0, 3.0])
+    b = np.array([1.1, 2.0, 3.0])
+    assert tree_mapped_np_all_close(a, b) is False
+
+
+def test_tree_mapped_np_all_close_simple_tuples_close():
+    a = (np.array([1.0, 2.0]), np.array([3.0, 4.0]))
+    b = (a[0] + 1e-05, a[1] + 1e-05)
+
+    assert tree_mapped_np_all_close(a, b) is True
+    assert tree_mapped_np_all_close(a, b, rtol=1e-06) is False
+
+
+def test_tree_mapped_np_all_close_nested_tuples():
+    a = (np.array([1.0]), (np.array([2.0]), np.array([3.0])))
+    b = (np.array([1.0]), (np.array([2.0]), np.array([3.0])))
+    assert tree_mapped_np_all_close(a, b) is True
+
+    c = (np.array([1.0]), (np.array([2.1]), np.array([3.0])))
+    assert tree_mapped_np_all_close(a, c) is False
```

Comment:
```suggestion
    c = (np.array([1.0]), (np.array([2.1]), np.array([3.0])))
    
    assert tree_mapped_np_all_close(a, b) is True
    assert tree_mapped_np_all_close(a, c) is False
```
Just as a personal preference